src/main/java/core/dao/manufacturer/ManufacturerDaoJdbc.java

```diff
@@ -0,0 +1,130 @@
+package core.dao.manufacturer;
+
+import core.lib.Dao;
+import core.model.DataProcessingException;
+import core.model.Manufacturer;
+import core.utils.ConnectionUtils;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoJdbc implements ManufacturerDao {
+    @Override
+    public Manufacturer add(Manufacturer manufacturer) {
+        String insert = "INSERT INTO manufacturers(name, country)"
+                        + " VALUES(?, ?);";
+        
+        try (Connection con = ConnectionUtils.getConnection()) {
+            PreparedStatement insertStatement = con.prepareStatement(insert,
+                    Statement.RETURN_GENERATED_KEYS);
+            insertStatement.setString(1, manufacturer.getName());
+            insertStatement.setString(2, manufacturer.getCountry());
+            insertStatement.executeUpdate();
+            
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject("id", Long.class));
+            }
+            insertStatement.close();
```

Comment:
statement won't be closed if exception occures, let's move it to try with resources, same with other methods

Reply:
Nice, thanks